src/widgets/block.rs

```diff
@@ -134,33 +143,90 @@ impl<'a> Default for Block<'a> {
 }
 
 impl<'a> Block<'a> {
+    /// # Example
+    /// ```
+    /// # use ratatui::widgets::Block;
+    /// # use ratatui::layout::Alignment;
+    /// # use ratatui::title::{Position, Title};
+    ///Block::default()
+    ///    .title(Title::default().content("Left").alignment(Alignment::Left))
+    ///    .title(
+    ///        Title::default()
+    ///            .content("Center")
+    ///            .alignment(Alignment::Center),
+    ///    )
+    ///    .title(
+    ///        Title::default()
+    ///            .content("Right")
+    ///            .alignment(Alignment::Right),
+    ///    )
+    ///    .title(
+    ///        Title::default()
+    ///            .content("Left")
+    ///            .alignment(Alignment::Left)
+    ///            .position(Position::Bottom),
+    ///    )
+    ///    .title(
+    ///        Title::default()
+    ///            .content("Center")
+    ///            .alignment(Alignment::Center)
+    ///            .position(Position::Bottom),
+    ///    )
+    ///    .title(
+    ///        Title::default()
+    ///            .content("Right")
+    ///            .alignment(Alignment::Right)
+    ///            .position(Position::Bottom),
+    ///    );
+    ///```
     pub fn title<T>(mut self, title: T) -> Block<'a>
     where
-        T: Into<Line<'a>>,
+        T: Into<Title<'a>>,
     {
-        self.title = Some(title.into());
+        self.titles.get_or_insert(Vec::new()).push(title.into());
         self
     }
-
     #[deprecated(
         since = "0.10.0",
         note = "You should use styling capabilities of `text::Line` given as argument of the `title` method to apply styling to the title."
     )]
     pub fn title_style(mut self, style: Style) -> Block<'a> {
```

Comment:
For the first code, we could have a `Titles` struct that would help developers deal with multiple title styling, alignment, and position. See [this comment](https://github.com/tui-rs-revival/ratatui/pull/232/files/e8c6464357cd0e41b668629028402ce633e2e237#r1225856916).

For the second code block, I'm not sure we should promote usage of deprecated functions. That might only lead us to work on old code instead of the new solutions we found. This is also the reason, I made `title_alignment` and `title_on_bottom` act on the first title only: to not promote their usage.

Reply:
Following from the previous. This would be removing a deprecation for a good reason. The behavior remains the same as it was originally (`Block::default().title("foo").title_style(s)` just works), while acting fine on multiple styles. I don't think the second block is good at all btw, but it highlights that we should note that this works as a default for all styles in the docs.

Reply:
I'm not sure to have understood. Do you want the `title_style` method to act on all titles thus overriding previous `title_style` method invocation (in the second block, it would mean the only modifier is italic) or do you want `title_style` to work on the currently set titles (in the second block, it would mean title A is just Bold, and title B is just Italic).

Reply:
To be more consistent with `title_position` and `title_alignment`, I made act on all the titles of the block. In the event a title already had some style, this method will [patch the style](https://docs.rs/ratatui/0.21.0/ratatui/text/struct.Line.html#method.patch_style) onto it.